src/openms/source/ANALYSIS/ID/PeptideProteinResolution.cpp

```diff
@@ -49,15 +49,15 @@ namespace OpenMS
   std::ostream& operator<< (std::ostream& os, const ConnectedComponent& conn_comp)
     {
       os << "Proteins: ";
-      for (std::set<Size>::iterator prot_it = conn_comp.prot_grp_indices.begin();
+      for (std::set<Size>::const_iterator prot_it = conn_comp.prot_grp_indices.begin();
```

Comment:
set<T>::const_iterator and set<T>::iterator may be implemented using the same type.
There is actually a longer story behind set iterators (including a defect and a change to the C++ standard)
long story short: std::set::iterator is now required to be read-only.